### DISCUSSION TITLE
How to create/get a bean of DataFetchingEnvironment with RequestScope?

### DISCUSSION THREAD

**Discussion by danielborges93** [graphql-java-kickstart/graphql-spring-boot#438] [Q&A]
I want to get a instance of `DataFetchingEnvironment` per scope as a bean in my code.

My motivation is to get the request authorization header per each request without passing the `DataFetchingEnvironment` as a resolver method param. What is the best way to do this?

Thanks!

**Comment by aymanmoukhtar** [comment]
I have the exact same issue, would like to access request information in a layered architecture, without having to pass it from the resolver through each method, maybe by injecting it through some bean in the level i want, is that possible?

**Comment by neerajsu** [comment]
@danielborges93 @aymanmoukhtar : I haven't tested it, but have you tried creating a bean with Session scope and manipulate/set the POJO variables at the base query level (once), based on DataFetchingEnvironment and inject it wherever else you need? That should provide you exactly what you need.

**Comment by oliemansm** [comment]
@danielborges93 @aymanmoukhtar 
Assuming both use cases are security related then I'd advise to add Spring Security to the mix. That's what we use in all our projects. A custom Spring Security filter parses and authorizes the `Authorization` request header even before it hits the GraphQL resolvers. This means that in the resolver methods but even anywhere down the line you can get a hold of the authenticated principal using default Spring Security mechanisms, e.g. `SecurityContextHolder.getContext().getAuthentication()`.

**Reply by kasaisrom** [reply]
How about the case when it's not security related? Say we want to get locale header from GraphQL request and pass it to resolvers via some global configuration?

**Reply by oliemansm** [reply]
In that case you should try the approach suggested by @neerajsu: https://github.com/graphql-java-kickstart/graphql-spring-boot/discussions/438#discussioncomment-225856.
